pkg/kubelet/cm/dra/healthinfo.go

```diff
@@ -0,0 +1,195 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dra
+
+import (
+	"encoding/json"
+	"os"
+	"sync"
+	"time"
+
+	"k8s.io/klog/v2"
+	"k8s.io/kubernetes/pkg/kubelet/cm/dra/state"
+)
+
+const (
+	healthTimeout = 30 * time.Second
+)
+
+// healthInfoCache is a cache of known device health.
+type healthInfoCache struct {
+	sync.RWMutex
+	HealthInfo *state.DevicesHealthMap
+	stateFile  string
+}
+
+// newHealthInfoCache creates a new cache, loading from a checkpoint if present.
+func newHealthInfoCache(stateFile string) (*healthInfoCache, error) {
+	cache := &healthInfoCache{
+		HealthInfo: &state.DevicesHealthMap{},
+		stateFile:  stateFile,
+	}
+	if err := cache.loadFromCheckpoint(); err != nil {
+		klog.Background().V(3).Info("Failed to load health checkpoint, proceeding with empty cache", "err", err)
```

Comment:
V3 seems to be too low. Wouldn't even be a default in many cases. Perhaps just info is OK

```suggestion
		klog.Background().Info("Failed to load health checkpoint, proceeding with empty cache", "err", err)
```